optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/rotation/TimeBucket.java

```diff
@@ -0,0 +1,210 @@
+            }
+            rotationDay = rotationDay.plus(1);
+        }
+    }
+    
+    public TimeBucket(Integer tenantId, Spot spot, LocalTime startTime, LocalTime endTime,
+                      Set<Skill> additionalSkillSet,
+                      Set<DayOfWeek> repeatOnDaySet, List<Seat> seatList) {
+        super(tenantId);
+        this.spot = spot;
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.additionalSkillSet = additionalSkillSet;
+        this.repeatOnDaySet = repeatOnDaySet;
+        this.seatList = new ArrayList<>(seatList);
+    }
+    
+    
+    public Spot getSpot() {
+        return spot;
+    }
+    
+    
+    public void setSpot(Spot spot) {
+        this.spot = spot;
+    }
+
+    public LocalTime getStartTime() {
+        return startTime;
+    }
+
+    
+    public void setStartTime(LocalTime startTime) {
+        this.startTime = startTime;
+    }
+
+    
+    public LocalTime getEndTime() {
+        return endTime;
+    }
+
+    
+    public void setEndTime(LocalTime endTime) {
+        this.endTime = endTime;
+    }
+
+    
+    public Set<Skill> getAdditionalSkillSet() {
+        return additionalSkillSet;
+    }
+
+    
+    public void setAdditionalSkillSet(Set<Skill> additionalSkillSet) {
+        this.additionalSkillSet = additionalSkillSet;
+    }
+
+    
+    public Set<DayOfWeek> getRepeatOnDaySet() {
+        return repeatOnDaySet;
+    }
+
+    
+    public void setRepeatOnDaySet(Set<DayOfWeek> repeatOnDaySet) {
+        this.repeatOnDaySet = repeatOnDaySet;
+    }
+
+    
+    public List<Seat> getSeatList() {
+        return seatList;
+    }
+
+    
+    public void setSeatList(List<Seat> seatList) {
+        this.seatList = seatList;
+    }
+
+    public void setValuesFromTimeBucket(TimeBucket updatedTimeBucket) {
+        setSpot(updatedTimeBucket.getSpot());
+        setStartTime(updatedTimeBucket.getStartTime());
+        setEndTime(updatedTimeBucket.getEndTime());
+        setAdditionalSkillSet(updatedTimeBucket.getAdditionalSkillSet());
+        setRepeatOnDaySet(updatedTimeBucket.getRepeatOnDaySet());
+        setSeatList(updatedTimeBucket.getSeatList());   
+    }
+    
+    public Optional<Shift> createShiftForOffset(LocalDate startDate, int offset, ZoneId zoneId,
+                                                boolean defaultToRotationEmployee) {
+        Optional<Seat> maybeSeat = seatList.stream().filter(seat -> seat.getDayInRotation() == offset).findAny();
+        
+        if (!maybeSeat.isPresent()) {
+            return Optional.empty();
+        }
+        Seat seat = maybeSeat.get();
+        LocalDateTime startDateTime = startDate.atTime(getStartTime());
+        LocalDate endDate;
+        
+        if (getStartTime().isBefore(getEndTime())) {
+            // Ex: 9am-5pm
+            endDate = startDate;
+        } else {
+            // Ex: 9pm-6am
+            endDate = startDate.plusDays(1);
+        }
```

Comment:
LocalDate endDate = (getStartTime().isBefore(getEndTime())) ?
       startDate  :
       startDate.plusDays(1);

Nicer, isn't it?